src/core/utils/helpers/list-mapper.ts

```diff
@@ -0,0 +1,161 @@
+import { LoanV2, RenewedLoanV2 } from "../../fbs/model";
+import { FaustId } from "../types/ids";
+import { store } from "../../store";
+import { GetMaterialManifestationQuery } from "../../dbc-gateway/generated/graphql";
+import { BasicDetailsType } from "../types/basic-details-type";
+import { Product, Loan } from "../../publizon/model";
+import { LoanType } from "../types/loan-type";
+
+// Creates a "by author, author and author"-string
+// String interpolation todo?
+export const getContributors = (creators: string[] | undefined) => {
+  const {
+    text: { data: texts }
+  } = store.getState();
+
+  let returnContentString = "";
+  if (creators && creators.length > 0) {
+    if (creators.length === 1) {
+      returnContentString = `${texts.materialByAuthorText} ${creators.join(
+        ", "
+      )}`;
+    } else {
+      returnContentString = `${texts.materialByAuthorText} ${creators
+        .slice(0, -1)
+        .join(", ")} ${texts.materialAndAuthorText} ${creators.slice(-1)}`;
+    }
+  }
+  return returnContentString;
+};
+
+//
+function getYearFromDataString(date?: string) {
```

Comment:
This is the same case as above why making date optional when the whole functionality depends on the argument?

Reply:
Imma change both